pkg/store/store.go

```diff
@@ -85,6 +85,45 @@ func (s *MsiKeyVaultStore) GetCredentialsObject(ctx context.Context, secretName
 	return &SecretResponse{CredentialsObject: credentialsObject, Properties: secretProperties}, nil
 }
 
+// Get a deleted credentials object from the key vault using the specified secret name.
+func (s *MsiKeyVaultStore) GetDeletedCredentialsObject(ctx context.Context, secretName string) (*SecretResponse, error) {
```

Comment:
Let's create a separate struct called `DeletedSecretResponse` to make it clear this is a deleted secret response.

Reply:
Added both DeletedSecretResponse and DeletedSecretProperties